examples/AcidRain.hs

```diff
@@ -31,39 +32,46 @@ userAction = Stream.repeatM $ liftIO askUser
             "quit"   -> return Quit
             _        -> putStrLn "Type potion or harm or quit" >> askUser
 
-acidRain :: MonadAsync m => SerialT m Event
+acidRain :: (MonadIO m, MonadAsync m) => Stream m Event
 acidRain =
-      Stream.repeatM (liftIO $ return $ Harm 1) -- AsyncT m Event
-    & Stream.constRate 1                        -- AsyncT m Event
-    & Stream.fromAsync                          -- SerialT m Event
+    Stream.repeat (liftIO $ return $ Harm 1)      -- Stream m Event
+        & Concur.parSequence (Concur.maxRate 1)
 
 data Result = Check | Done
 
-runEvents :: (MonadAsync m, MonadState Int m) => SerialT m Result
-runEvents = do
-    event <- userAction `Stream.parallel` acidRain -- SerialT m Event
-    case event of
-        Harm n -> modify (\h -> h - n) >> return Check
-        Heal n -> modify (\h -> h + n) >> return Check
-        Quit -> return Done
+runEvents :: (MonadIO m, MonadAsync m) => IORef Int -> Stream m Result
+runEvents ref =
+    let events = Concur.eval $ Concur.append2 userAction acidRain
+        e2 = Stream.takeEndBy (== Quit ) events     -- Stream m Event
+      in Stream.mapM modCount e2                    -- Stream m Result
+
+    where
+
+    modCount ev =
+        case  ev of
+            Harm n -> liftIO $ modifyIORef ref (\h -> h - n) >> return Check
+            Heal n -> liftIO $ modifyIORef ref (+ n) >> return Check
+            Quit -> return Done
```

Comment:
Can we keep it the same as old code and just replace `parallel` with `parEager` or `parConcatList (eager True)`?

Reply:
We wanted to use the updated Stream.